### PR TITLE
Add some basics about troubleshooting the Collector

### PR DESCRIPTION
**Pull request by slernersplunk**
Added some helpful lines for troubleshooting the collector...

**Description:** <Describe what has changed. 
Added some documentation

**Documentation:** < Describe the documentation added.>
basics about troubleshooting the Collector

### CODE REVIEW

**Review by bogdandrutu — APPROVED** [review]

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -194,6 +194,14 @@ Collector as it runs. This is an advanced use-case that should not be needed in
 
 ## Common Issues
 
+To see logs for the Collector:
+
+On a Linux systemd system, logs can be found using `journalctl`:  
+`journalctl | grep otel`
```
Comment (bogdandrutu) [inline docs/troubleshooting.md:200]: why here otel then 2 lines bellow otelcol?

Comment (stevelerner) [reply docs/troubleshooting.md:200]: probably due to multitasking working with customer while at same time making suggestions- feel free to fix as you like...

Comment (slernersplunk) [reply docs/troubleshooting.md:200]: feel free to pick whichever you'd prefer...

Comment (jpkrohling) [reply docs/troubleshooting.md:200]: If it is started as a systemd unit, you can use the `-u` flag to `journalctl` instead of grepping for otel.

Comment (slernersplunk) [reply docs/troubleshooting.md:200]: @jpkrohling what is the full journalctl command that you recommend so I can test it with users?

Comment (jpkrohling) [reply docs/troubleshooting.md:200]: Here's an example when using the otel-collector installed via the RPM package, with the expected output:

```console
$ journalctl -u otel-collector
-- Logs begin at Wed 2021-04-07 14:00:18 CEST, end at Wed 2021-04-28 14:52:07 CEST. --
Apr 28 14:51:41 guarana systemd[1]: Started OpenTelemetry Collector.
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.754+0200        info        service/application.go:281        Starting OpenTelemetry Collector...        {"Version": "v0.25.0", "GitHash": "d93c9d29", "NumCPU": 12}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.754+0200        info        service/application.go:189        Setting up own telemetry...
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.766+0200        info        service/telemetry.go:98        Serving Prometheus metrics        {"address": ":8888", "level": 0, "service.instance.id": "815b7908-712c-427c-89d5-dd>
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.766+0200        info        service/application.go:224        Loading configuration...
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        service/application.go:240        Applying configuration...
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/exporters_builder.go:274        Exporter was built.        {"kind": "exporter", "exporter": "logging"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/pipelines_builder.go:204        Pipeline was built.        {"pipeline_name": "metrics", "pipeline_datatype": "metrics"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/pipelines_builder.go:204        Pipeline was built.        {"pipeline_name": "traces", "pipeline_datatype": "traces"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:230        Receiver was built.        {"kind": "receiver", "name": "jaeger", "datatype": "traces"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:230        Receiver was built.        {"kind": "receiver", "name": "zipkin", "datatype": "traces"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:230        Receiver was built.        {"kind": "receiver", "name": "prometheus", "datatype": "metrics"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:230        Receiver was built.        {"kind": "receiver", "name": "otlp", "datatype": "traces"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:230        Receiver was built.        {"kind": "receiver", "name": "otlp", "datatype": "metrics"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:230        Receiver was built.        {"kind": "receiver", "name": "opencensus", "datatype": "traces"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:230        Receiver was built.        {"kind": "receiver", "name": "opencensus", "datatype": "metrics"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        service/service.go:155        Starting extensions...
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/extensions_builder.go:53        Extension is starting...        {"kind": "extension", "name": "health_check"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        healthcheckextension/healthcheckextension.go:41        Starting health_check extension        {"kind": "extension", "name": "health_check", "config">
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/extensions_builder.go:59        Extension started.        {"kind": "extension", "name": "health_check"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/extensions_builder.go:53        Extension is starting...        {"kind": "extension", "name": "pprof"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        pprofextension/pprofextension.go:79        Starting net/http/pprof server        {"kind": "extension", "name": "pprof", "config": {"TypeVal":"pprof">
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/extensions_builder.go:59        Extension started.        {"kind": "extension", "name": "pprof"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/extensions_builder.go:53        Extension is starting...        {"kind": "extension", "name": "zpages"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        zpagesextension/zpagesextension.go:45        Host's zPages not available        {"kind": "extension", "name": "zpages"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        zpagesextension/zpagesextension.go:55        Starting zPages extension        {"kind": "extension", "name": "zpages", "config": {"TypeVal":"zpages",>
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/extensions_builder.go:59        Extension started.        {"kind": "extension", "name": "zpages"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        service/service.go:200        Starting exporters...
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/exporters_builder.go:92        Exporter is starting...        {"kind": "exporter", "name": "logging"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/exporters_builder.go:97        Exporter started.        {"kind": "exporter", "name": "logging"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        service/service.go:205        Starting processors...
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/pipelines_builder.go:51        Pipeline is starting...        {"pipeline_name": "metrics", "pipeline_datatype": "metrics"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/pipelines_builder.go:62        Pipeline is started.        {"pipeline_name": "metrics", "pipeline_datatype": "metrics"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/pipelines_builder.go:51        Pipeline is starting...        {"pipeline_name": "traces", "pipeline_datatype": "traces"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/pipelines_builder.go:62        Pipeline is started.        {"pipeline_name": "traces", "pipeline_datatype": "traces"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        service/service.go:210        Starting receivers...
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:70        Receiver is starting...        {"kind": "receiver", "name": "jaeger"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        static/strategy_store.go:201        No sampling strategies provided or URL is unavailable, using defaults        {"kind": "receiver", "name": "jaege>
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:75        Receiver started.        {"kind": "receiver", "name": "jaeger"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:70        Receiver is starting...        {"kind": "receiver", "name": "zipkin"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:75        Receiver started.        {"kind": "receiver", "name": "zipkin"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.767+0200        info        builder/receivers_builder.go:70        Receiver is starting...        {"kind": "receiver", "name": "prometheus"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.950+0200        info        builder/receivers_builder.go:75        Receiver started.        {"kind": "receiver", "name": "prometheus"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.950+0200        info        builder/receivers_builder.go:70        Receiver is starting...        {"kind": "receiver", "name": "otlp"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.950+0200        info        otlpreceiver/otlp.go:94        Starting GRPC server on endpoint 0.0.0.0:4317        {"kind": "receiver", "name": "otlp"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.950+0200        info        otlpreceiver/otlp.go:137        Setting up a second GRPC listener on legacy endpoint 0.0.0.0:55680        {"kind": "receiver", "name": "otlp"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.950+0200        info        otlpreceiver/otlp.go:94        Starting GRPC server on endpoint 0.0.0.0:55680        {"kind": "receiver", "name": "otlp"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.950+0200        info        otlpreceiver/otlp.go:112        Starting HTTP server on endpoint 0.0.0.0:55681        {"kind": "receiver", "name": "otlp"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.950+0200        info        builder/receivers_builder.go:75        Receiver started.        {"kind": "receiver", "name": "otlp"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.950+0200        info        builder/receivers_builder.go:70        Receiver is starting...        {"kind": "receiver", "name": "opencensus"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.950+0200        info        builder/receivers_builder.go:75        Receiver started.        {"kind": "receiver", "name": "opencensus"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.950+0200        info        healthcheck/handler.go:128        Health Check state change        {"kind": "extension", "name": "health_check", "status": "ready"}
Apr 28 14:51:41 guarana otelcol[262026]: 2021-04-28T14:51:41.950+0200        info        service/application.go:201        Everything is ready. Begin running and processing data.

```

**Comment by slernersplunk** [comment]
@jpkrohling That doesn't work for all contribs and there are many... which is why I suggested grep...

`

ubuntu@ip-172-31-50-89:~$ journalctl -u otel-collector
-- Logs begin at Wed 2021-04-28 12:14:42 UTC, end at Wed 2021-04-28 14:48:14 UTC. --
-- No entries --

ubuntu@ip-172-31-50-89:~$ journalctl | grep otel
Apr 28 12:24:51 ip-172-31-50-89 k3s[4860]: I0428 12:24:51.764108    4860 event.go:291] "Event occurred" object="default/splunk-otel-collector-1619612690-k8s-cluster-receiver" kind="Deployment" apiVersion="apps/v1" type="Normal" reason="ScalingReplicaSet" message="Scaled up replica set splunk-otel-collector-1619612690-k8s-cluster-receiver-85c5c99dc8 to 1"`

**Comment by jpkrohling** [comment]
The log entry you've shown isn't one that is produced by the OpenTelemetry Collector, but by Kubernetes (k3s):

```
Apr 28 12:24:51 ip-172-31-50-89 k3s[4860]: I0428 12:24:51.764108    4860 event.go:291] "Event occurred" object="default/splunk-otel-collector-1619612690-k8s-cluster-receiver" kind="Deployment" apiVersion="apps/v1" type="Normal" reason="ScalingReplicaSet" message="Scaled up replica set splunk-otel-collector-1619612690-k8s-cluster-receiver-85c5c99dc8 to 1"
```

**Comment by slernersplunk** [comment]
journalctl is a fine place to get collector k8s errors as well-- but here is a non k8s example:

```
Apr 27 16:09:02 eu4redact.corp.redacted.com otelcol[14540]: Error: cannot setup extensions: failed to start extensions: failed to bind to address :6060: listen tcp :6060: bind: address already in use
```

That only appears via `journalctl | grep` and not via a universal `-u` option.... and this is not related to k8s.

**Comment by jpkrohling** [comment]
>journalctl is a fine place to get collector k8s errors as well

Are you able to get the logs for a Kubernetes deployment via journalctl from the main host? That's cool, didn't know that. Would that include logs from deployments in other nodes too?

> That only appears via journalctl | grep and not via a universal -u option.... and this is not related to k8s.

When I'm wearing my sysadm hat, I prefer to get logs related only to the service I'm debugging, hence the `-u unitname` flag. Otherwise, I might get entries from other systems, which would possibly negatively affect my judgment. In any case, if you think the right thing is to indeed grep by otelcol, I'm fine with that.

**Comment by slernersplunk** [comment]
> Are you able to get the logs for a Kubernetes deployment via journalctl from the main host? That's cool, didn't know that. Would that include logs from deployments in other nodes too?

Didn't say its for everything- but its a perfectly fine place to get whats available.

> > That only appears via journalctl | grep and not via a universal -u option.... and this is not related to k8s.
> 
> When I'm wearing my sysadm hat, I prefer to get logs related only to the service I'm debugging, hence the `-u unitname` flag. Otherwise, I might get entries from other systems, which would possibly negatively affect my judgment. In any case, if you think the right thing is to indeed grep by otelcol, I'm fine with that.

All that is true- however given that each contrib has its own full name so `journalctl` can't be a single universal recommendation, we currently don't have any other reliable way to suggest troubleshooting hence this option. Goal is fast, easy, and universal. I've worked with several people trying it out and this one step, if in the docs, would have saved them a lot of time when troubleshooting.

**Comment by jpkrohling** [comment]
>All that is true- however given that each contrib has its own full name

Do you mean "distribution" here? I would expect distribution owners to either use the same unit name as the core or to let their users aware of the deviations from the standard. I believe your change is indeed an improvement, but it also has the potential for creating confusion.

If you did men "the contrib'' distribution, then its unit name is [`otel-contrib-collector`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1e8d16e61ba73f53a4e22ff26f7ef741d66b14c1/internal/buildscripts/packaging/fpm/otel-contrib-collector.service), so the doc here could state which units to use for which binaries.

**Comment by slernersplunk** [comment]
> Do you mean "distribution" here?

That is a synonym yes. 

> I would expect distribution owners to either use the same unit name as the core or to let their users aware of the deviations from the standard.

There is no guarantee of this...

> I believe your change is indeed an improvement, but it also has the potential for creating confusion.

It is just a simple helper for troubleshooting quickly... we dont have a lot of troubleshooting guidance and that particular one would (and has) saved people a lot of time...

We can always firm this up in the future but since our goal is mass acceptance lets give the people a little help...

> 
> If you did men "the contrib'' distribution, then its unit name is [`otel-contrib-collector`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/1e8d16e61ba73f53a4e22ff26f7ef741d66b14c1/internal/buildscripts/packaging/fpm/otel-contrib-collector.service), so the doc here could state which units to use for which binaries.

**Comment by slernersplunk** [comment]
can we get this approved? thx

**Comment by bogdandrutu** [comment]
@jpkrohling can you approve?

@slernersplunk can you please rebase so newly added test pass?

File: docs/troubleshooting.md (hunk 1 above)
Comment (tigrannajaryan) [reply docs/troubleshooting.md:200]: > feel free to pick whichever you'd prefer...

@slernersplunk Please make it consistent. I think `otelcol` fits better.

**Review by tigrannajaryan — APPROVED** [review]

**Review by slernersplunk — COMMENTED** [review]
per req changed both examples to otelcol

File: docs/troubleshooting.md (hunk 1 above)
Comment (slernersplunk) [reply docs/troubleshooting.md]: done

**Pull request merged**